### ISSUE TITLE
Wrong execution order in SuiteManager::addToSuite

### ISSUE THREAD

**Issue by zemlyanskiy** [Codeception/Codeception#2386]
SuiteManager.php Line 116:
```
if ($test instanceof TestCase) {
            if (!$this->isCurrentEnvironment($test->getEnvironment())) {
                return; // skip tests from other environments
            }
        }
        if ($test instanceof ScenarioDriven) {
            $test->preload();
        }
```

It's wrong, because $test->getEnvironment() need initialized $test->scenario for correct work

This is correct order:

```
 if ($test instanceof ScenarioDriven) {
            $test->preload();
        }
if ($test instanceof TestCase) {
            if (!$this->isCurrentEnvironment($test->getEnvironment())) {
                return; // skip tests from other environments
            }
        }
```

**Comment by DavertMik** [comment]
yes, we should have fixed it in current 2.1 branch but not released it yet. Please check that this fix is inside repo already

**Comment by zemlyanskiy** [comment]
thank you!
